xbmc/guilib/XBTFReader.cpp

```diff
@@ -31,16 +31,16 @@
 #include "PlatformDefs.h"
 
 #define READ_STR(str, size, file) \
-  if (!fread(str, size, 1, file)) \
+  if (1 != fread(str, size, 1, file)) \
```

Comment:
It shouldnt be, docs says it returns count of elements read and count is always 1 so any return except 1 should be a failure